### PR TITLE
Super modernization

### PR DESCRIPTION
**Pull request by chris-pardy**
Super Modernization. Also note that this is a open source plugin so still to-do are finish the deployment script, and update the Readme

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
+{
+  "compilerOptions": {
```
Comment (RoboCafaz) [inline tsconfig.json:2]: Can we axe all these commented-out config values? I'm fine with leaving the comments for the included fields and the category comments.

Comment (chris-pardy) [reply tsconfig.json:2]: ok

**Comment by RoboCafaz** [comment]
Any particular reason you switched to rollup? Is this something we might want to incorporate back into our other stacks? Is this more exploratory since this is a periphery library that won't get touched often?

**Comment by chris-pardy** [comment]
> Any particular reason you switched to rollup? Is this something we might want to incorporate back into our other stacks? Is this more exploratory since this is a periphery library that won't get touched often?

Rollup seems to have basically taken over as the defacto standard for bundling libraries. The main advantage it has is that it produces a single file optimized for ES Modules. Doing that bundling let's us think much more about what is an implementation detail vs. what is exported. (anything not exported from index is effectively "private" to our library)

**Comment by chris-pardy** [comment]
Before I merge this I want to get travis building it and ready to deploy it. So looking for other reviews.

**Comment by RoboCafaz** [comment]
> Before I merge this I want to get travis building it and ready to deploy it. So looking for other reviews.

Since Travis isn't configured yet, do you want to do it on codebuild?

**Comment by chris-pardy** [comment]
> > Before I merge this I want to get travis building it and ready to deploy it. So looking for other reviews.
> 
> Since Travis isn't configured yet, do you want to do it on codebuild?

I'm going to try to get this setup on travis such that someone outside of the organization can view the build logs but not edit the configuration. I don't think there's any practical way to do that with Code Build.

**Pull request merged**